Core/Core/Use Cases/UploadManager.swift

```diff
@@ -348,8 +352,12 @@ open class UploadManager: NSObject, URLSessionDelegate, URLSessionTaskDelegate,
                 file.localFileURL = nil
             }
             try? context.save()
-            if error != nil, case let .submission(courseID, assignmentID, _)? = file.context {
-                sendFailedNotification(courseID: courseID, assignmentID: assignmentID)
+            if error != nil {
+                if case let .submission(courseID, assignmentID, _)? = file.context {
+                    sendFailedNotification(courseID: courseID, assignmentID: assignmentID)
+                } else {
+                    sendFailedNotification()
```

Comment:
I'm not sure about this, but it seems like in other error handling places we call this when there is no submission context.